src/workerd/util/sqlite.c++

```diff
@@ -992,6 +998,10 @@ SqliteDatabase::Query::~Query() noexcept(false) {
       sqlite3_clear_bindings(&statement);
     }
   }
+  // Update Query stats only if db is not reset
+  if (maybeStatement != kj::none) {
+    db.sqliteObserver.updateQueryStats(getRowsRead(), getRowsWritten());
```

Comment:
I would have thought that the `sqlite3_reset()` call on line 994 would affect the statistics here.  I guess it doesn't?

Reply:
No, I meant that the calls to getRowsRead()/getRowsWritten() would have been affected by the reset, since those call into `sqlite3_stmt_status(statement, LIBSQL_STMTSTATUS_ROWS_WRITTEN, 0);`.  It seems `sqlite3_reset()` doesn't touch those values.